manifest/package.go

```diff
@@ -90,6 +98,87 @@ func (r *manifestConverter) validateManifest(manif *Manifest) error {
 	return nil
 }
 
+func handleUseModules(pkg *pbsubstreams.Package, manif *Manifest) error {
+	packageModulesMapping := make(map[string]*pbsubstreams.Module)
+	for _, module := range pkg.Modules.Modules {
+		packageModulesMapping[module.Name] = module
+	}
+
+	for _, manifestModule := range manif.Modules {
+		if manifestModule.Use != "" {
+			usedModule, found := packageModulesMapping[manifestModule.Use]
+			if !found {
+				return fmt.Errorf("module %q: use module %q not found", manifestModule.Name, manifestModule.Use)
+			}
+			moduleWithUse := packageModulesMapping[manifestModule.Name]
+
+			if err := checkEqualInputs(moduleWithUse, usedModule, manifestModule, packageModulesMapping); err != nil {
+				return fmt.Errorf("checking inputs for module %q: %w", manifestModule.Name, err)
+			}
+
+			moduleWithUse.BinaryIndex = usedModule.BinaryIndex
+			moduleWithUse.BinaryEntrypoint = usedModule.BinaryEntrypoint
+
+			if moduleWithUse.InitialBlock == 0 {
```

Comment:
Remove these 3 lines completely: since we cannot differentiate between "0" and "nil", let's force the user to always set an initialBlock, so we don't override with the usedModule.